desktop/src/views/Pro/CreateWorkspace/UpdateWorkspace.tsx

```diff
@@ -26,10 +27,28 @@ export function UpdateWorkspace({ instance, template }: TUpdateWorkspaceProps) {
   const { host, client } = useProContext()
   const [globalError, setGlobalError] = useState<Failed | null>(null)
 
+  const { data: templates, isLoading: isTemplatesLoading } = useTemplates()
+
+  const presets = templates?.presets
+
+  const [presetId, setPresetId] = useState<string | undefined>(instance.spec?.presetRef?.name)
+
+  useEffect(() => {
+    setPresetId(instance.spec?.presetRef?.name)
+  }, [instance.spec?.presetRef?.name])
```

Comment:
also could go into the `useMemo`?

Reply:
Arguably you could find a way to merge this one with its following `useMemo` but I'd question why you would want that.

As it is, this effect serves one single purpose: Synchronize the internal state with any new changes pushed from the parent component. Neat and tidy.

The following `useMemo` resolves the internal preset ID - which can also change from other sources(!) - to the full preset data, without any side effects.

This are distinct purposes, I think we should keep it like it is.